applications/nrf_desktop/src/modules/ble_qos.c

```diff
@@ -0,0 +1,741 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <device.h>
+#include <drivers/uart.h>
+#include <misc/byteorder.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+
+#include "ble_controller_hci_vs.h"
+
+#include "chmap_filter.h"
+
+#define MODULE ble_qos
+#include "module_state_event.h"
+#include "ble_event.h"
+#include "config_event.h"
+#include "hid_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_BLE_QOS_LOG_LEVEL);
+
+#define INVALID_BLACKLIST 0xFFFF
+
+#if CONFIG_DESKTOP_BLE_QOS_STATS_PRINTOUT_ENABLE
+#define THREAD_STACK_SIZE \
+	(CONFIG_DESKTOP_BLE_QOS_STACK_SIZE + \
+	CONFIG_DESKTOP_BLE_QOS_STATS_PRINT_STACK_SIZE)
+#else
+#define THREAD_STACK_SIZE CONFIG_DESKTOP_BLE_QOS_STACK_SIZE
+#endif
+
+#if CONFIG_DESKTOP_BLE_QOS_STATS_PRINTOUT_ENABLE
+# if DT_NORDIC_NRF_USBD_USBD_0_NUM_IN_ENDPOINTS < 4
+# error Too few USB IN Endpoints enabled.\
+Modify appropriate dts.overlay to increase num-in-endpoints to 4 or more
+# endif
+#endif
+
+#ifndef ROUNDED_DIV
+#define ROUNDED_DIV(a, b)  (((a) + ((b) / 2)) / (b))
+#endif /* ROUNDED_DIV */
+
+/* Lowest thread priority */
+#define THREAD_PRIORITY K_PRIO_PREEMPT((CONFIG_NUM_PREEMPT_PRIORITIES - 1))
+
+static K_THREAD_STACK_DEFINE(thread_stack, THREAD_STACK_SIZE);
+static struct k_thread thread;
+
+struct params_ble {
+	u16_t sample_count_min;
+	u8_t min_channel_count;
```

Comment:
putting `min_channel_count` just before `eval_max_count` will save 1 B. I know it ain't much but still :)

Reply:
Yes, if __packed attribute is removed. It's easier to retain the __packed attribute, such that future additions to these structs can be added to the end of the struct, and to the end of the corresponding format specifiers in the python script

Reply:
I have added comment above. Let's leave it as is.